src/Reshaping.jl

```diff
@@ -8,6 +8,8 @@ periodicity of a requested supercell. The columns of the ``3×3`` integer matrix
 original crystal lattice vectors.
 """
 function reshape_supercell(sys::System{N}, shape) where N
+    is_homogeneous(sys) || error("Cannot reshape system with inhomogeneous interactions.")
```

Comment:
The point of this is to prevent users from doing this? In other words, you have to know about `reshape_supercell_aux`.

Reply:
Yes. I assume we don't want people reshaping inhomogeneous systems in general. This PR makes an exception for LSWT, which is a very specialized use case.

Reply:
Update -- there's a bit more flexibility now. For example, with the latest commits, it becomes possible to call `repeat_periodically` on an inhomogeneous system.